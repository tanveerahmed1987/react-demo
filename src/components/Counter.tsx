import React from "react";
import { observer } from "mobx-react";
import store from "../store/CounterStore";
import { CounterWrapper, StyledText } from "./styles";

const Counter: React.FC = () => {
  return (
    <CounterWrapper>
      <button className="button button1" onClick={() => store.increment()}>
        Increment
      </button>

      <StyledText>{store.count}</StyledText>

      <button className="button button3" onClick={() => store.decrement()}>
        Decrement
      </button>
    </CounterWrapper>
  );
};

export default observer(Counter);
