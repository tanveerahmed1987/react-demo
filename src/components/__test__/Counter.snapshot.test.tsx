// components/__tests__/Counter.snapshot.test.tsx
import * as React from "react";
import Counter from "../Counter";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Counter />).toJSON();
  expect(tree).toMatchSnapshot();
});
