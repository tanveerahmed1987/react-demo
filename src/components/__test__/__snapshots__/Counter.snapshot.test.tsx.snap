// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdvvaa gUUmgX"
>
  <button
    className="button button1"
    onClick={[Function]}
  >
    Increment
  </button>
  <div
    className="sc-gsDJrp cvflqT"
  >
    0
  </div>
  <button
    className="button button3"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
